segment git: url parsing error when using "custom" / "non-standard" ssh user

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

I'm running a [gitea](https://gitea.io/) instance for private stuff and have an issue with the git segment. It shows the following error message instead of the icon, branch name, ...:

`: error calling url: parse "https://_gitea@src.example.com:user/repo": invalid port ":user" after host`

The url in .git/config looks like that:

`url = _gitea@src.example.com:user/repo.git`

When i change the url like this:

`url = git@src.example.com:user/repo.git`

the git segment works like expected. Seems to me, like there's a url parsing issue...

Gitea uses standard username **_gitea** which differs from many other service like Github, Gitlab, ... which are using **git** as ssh username.

Would be great, if you could take a look at this and find a solution for.

Thanks and best regards, Don.

### Theme

Night Owl

### What OS are you seeing the problem on?

macOS

### Which shell are you using?

bash

### Log output

```Shell
Version: 14.2.2

Segments:

ConsoleTitle(true)  -   0 ms - repo
os(true)            -   0 ms - ╭─
shell(true)         -   0 ms -   bash 
root(false)         -   0 ms -
path(true)          -   0 ms -        repo
git(true)           -  21 ms - : error calling url: parse "https://_gitea@src.example.com:user/repo": invalid port ":user" after host
executiontime(true) -   0 ms -  0ms
battery(true)       -  18 ms - 80 
aws(false)          -   0 ms -
az(false)           -   0 ms -
azfunc(false)       -   0 ms -
go(false)           -   0 ms -
kubectl(false)      -  48 ms -
perl(false)         -   0 ms -
python(false)       -   0 ms -
ruby(false)         -   0 ms -
rust(false)         -   0 ms -
time(true)          -   0 ms -   11:11:04
text(true)          -   0 ms - ╰─
exit(true)          -   0 ms - 

Run duration: 96.031449ms

Cache path: /Users/don/.cache/oh-my-posh

Config path: /Users/don/.omp/my-night-owl.omp.json

Logs:

[DEBUG] 11:11:04.482 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.482 shell.go:Getenv(XDG_CACHE_HOME) - 37.396µs
[TRACE] 11:11:04.482 shell_unix.go:CachePath() - 81.022µs
[DEBUG] 11:11:04.482 shell.go:Getenv:347 → /Users/don/.omp/my-night-owl.omp.json
[TRACE] 11:11:04.482 shell.go:Getenv(POSH_THEME) - 6.655µs
[TRACE] 11:11:04.482 shell.go:resolveConfigPath() - 11.722µs
[TRACE] 11:11:04.482 shell.go:Init() - 551.573µs
[TRACE] 11:11:04.482 shell.go:Flags() - 97ns
[TRACE] 11:11:04.484 config.go:loadConfig() - 1.279333ms
[TRACE] 11:11:04.484 shell.go:Flags() - 125ns
[DEBUG] 11:11:04.484 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.484 shell.go:Getenv(OMP_CACHE_DISABLED) - 3.543µs
[TRACE] 11:11:04.484 shell_unix.go:Root() - 1.138µs
[DEBUG] 11:11:04.484 shell.go:Shell:592 → process name: bash
[TRACE] 11:11:04.484 shell.go:Shell() - 129.724µs
[TRACE] 11:11:04.484 shell.go:ErrorCode() - 108ns
[DEBUG] 11:11:04.484 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.484 shell.go:Getenv(WSL_DISTRO_NAME) - 2.946µs
[TRACE] 11:11:04.484 shell_unix.go:IsWsl() - 5.991µs
[TRACE] 11:11:04.484 shell.go:GOOS() - 111ns
[DEBUG] 11:11:04.484 shell.go:Pwd:375 → NO DATA
[TRACE] 11:11:04.484 shell.go:Pwd() - 31.185µs
[TRACE] 11:11:04.484 shell.go:GOOS() - 49ns
[TRACE] 11:11:04.484 shell.go:GOOS() - 48ns
[TRACE] 11:11:04.484 shell.go:GOOS() - 48ns
[TRACE] 11:11:04.484 shell.go:GOOS() - 48ns
[TRACE] 11:11:04.484 shell.go:GOOS() - 49ns
[DEBUG] 11:11:04.484 shell.go:User:496 → don
[TRACE] 11:11:04.484 shell.go:User() - 3.059µs
[DEBUG] 11:11:04.484 shell.go:Host:508 → mb2-don.fritz.box
[TRACE] 11:11:04.484 shell.go:Host() - 16.103µs
[TRACE] 11:11:04.484 shell.go:GOOS() - 50ns
[TRACE] 11:11:04.484 shell.go:TemplateCache() - 261.232µs
[TRACE] 11:11:04.484 shell.go:Flags() - 94ns
[DEBUG] 11:11:04.484 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.484 shell.go:Pwd() - 3.639µs
[TRACE] 11:11:04.484 shell.go:GOOS() - 65ns
[TRACE] 11:11:04.484 shell.go:GOOS() - 46ns
[TRACE] 11:11:04.484 shell.go:TemplateCache() - 137ns
[TRACE] 11:11:04.485 shell.go:TemplateCache() - 206ns
[TRACE] 11:11:04.485 shell.go:Shell() - 130ns
[DEBUG] 11:11:04.485 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.485 shell.go:Pwd() - 4.48µs
[TRACE] 11:11:04.485 shell.go:GOOS() - 70ns
[TRACE] 11:11:04.485 shell.go:Shell() - 114ns
[TRACE] 11:11:04.485 shell.go:Flags() - 73ns
[TRACE] 11:11:04.485 shell.go:TemplateCache() - 90ns
[TRACE] 11:11:04.485 shell.go:TemplateCache() - 177ns
[TRACE] 11:11:04.485 shell.go:Shell() - 65ns
[DEBUG] 11:11:04.485 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.485 shell.go:Pwd() - 4.046µs
[TRACE] 11:11:04.485 shell.go:GOOS() - 71ns
[TRACE] 11:11:04.485 shell_unix.go:Root() - 1.1µs
[DEBUG] 11:11:04.485 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.485 shell.go:Pwd() - 6.624µs
[TRACE] 11:11:04.485 shell.go:GOOS() - 51ns
[DEBUG] 11:11:04.485 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.485 shell.go:Pwd() - 2.914µs
[TRACE] 11:11:04.485 shell.go:Shell() - 92ns
[TRACE] 11:11:04.485 shell.go:Shell() - 51ns
[TRACE] 11:11:04.485 shell.go:GOOS() - 52ns
[TRACE] 11:11:04.485 shell.go:PathSeparator() - 78ns
[TRACE] 11:11:04.485 shell.go:GOOS() - 61ns
[TRACE] 11:11:04.485 shell.go:GOOS() - 56ns
[TRACE] 11:11:04.485 shell.go:GOOS() - 80ns
[TRACE] 11:11:04.485 shell.go:GOOS() - 49ns
[TRACE] 11:11:04.485 shell.go:PathSeparator() - 49ns
[TRACE] 11:11:04.485 shell.go:PathSeparator() - 59ns
[TRACE] 11:11:04.485 shell.go:GOOS() - 58ns
[TRACE] 11:11:04.485 shell.go:GOOS() - 61ns
[TRACE] 11:11:04.485 shell.go:PathSeparator() - 68ns
[TRACE] 11:11:04.485 shell.go:GOOS() - 59ns
[TRACE] 11:11:04.485 shell.go:GOOS() - 70ns
[TRACE] 11:11:04.485 shell.go:PathSeparator() - 60ns
[TRACE] 11:11:04.485 shell.go:GOOS() - 59ns
[TRACE] 11:11:04.485 shell.go:GOOS() - 57ns
[TRACE] 11:11:04.485 shell.go:PathSeparator() - 71ns
[TRACE] 11:11:04.485 shell.go:PathSeparator() - 47ns
[TRACE] 11:11:04.485 shell.go:PathSeparator() - 49ns
[TRACE] 11:11:04.485 shell.go:PathSeparator() - 49ns
[TRACE] 11:11:04.485 shell.go:PathSeparator() - 49ns
[DEBUG] 11:11:04.485 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.485 shell.go:Pwd() - 3.209µs
[DEBUG] 11:11:04.485 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.485 shell.go:Getenv(WSL_DISTRO_NAME) - 6.382µs
[TRACE] 11:11:04.485 shell_unix.go:IsWsl() - 9.524µs
[TRACE] 11:11:04.485 shell.go:StackCount() - 106ns
[TRACE] 11:11:04.485 shell_unix.go:DirIsWritable(/Users/don/src/example.com/repo) - 29.182µs
[TRACE] 11:11:04.485 shell.go:TemplateCache() - 121ns
[TRACE] 11:11:04.486 shell.go:TemplateCache() - 118ns
[TRACE] 11:11:04.486 shell.go:Shell() - 282ns
[DEBUG] 11:11:04.486 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.486 shell.go:Pwd() - 5.188µs
[TRACE] 11:11:04.486 shell.go:GOOS() - 71ns
[DEBUG] 11:11:04.486 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.486 shell.go:Getenv(WSL_DISTRO_NAME) - 5.017µs
[TRACE] 11:11:04.486 shell_unix.go:IsWsl() - 7.997µs
[TRACE] 11:11:04.486 shell_unix.go:IsWsl2() - 10.739µs
[TRACE] 11:11:04.486 shell.go:GOOS() - 48ns
[DEBUG] 11:11:04.486 shell.go:CommandPath:547 → /usr/local/bin/git
[TRACE] 11:11:04.486 shell.go:CommandPath(git) - 73.354µs
[TRACE] 11:11:04.486 shell.go:HasCommand(git) - 76.951µs
[DEBUG] 11:11:04.486 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.486 shell.go:Pwd() - 14.26µs
[TRACE] 11:11:04.486 shell.go:HasParentFilePath(.git) - 32.389µs
[TRACE] 11:11:04.486 shell.go:GOOS() - 71ns
[TRACE] 11:11:04.486 shell.go:GOOS() - 50ns
[TRACE] 11:11:04.486 shell.go:GOOS() - 48ns
[TRACE] 11:11:04.486 shell.go:GOOS() - 49ns
[TRACE] 11:11:04.486 shell.go:GOOS() - 49ns
[TRACE] 11:11:04.486 shell.go:GOOS() - 47ns
[DEBUG] 11:11:04.486 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.486 shell.go:Getenv(POSH_GIT_STATUS) - 3.045µs
[TRACE] 11:11:04.486 shell.go:GOOS() - 48ns
[DEBUG] 11:11:04.504 shell.go:RunCommand:526 ↓
    # branch.oid dd45ae9486b42daea3173d84b72a9a43a8b45a0a
    # branch.head master
    # branch.upstream origin/master
    # branch.ab +0 -0
[TRACE] 11:11:04.504 shell.go:RunCommand(git -C /Users/don/src/example.com/repo --no-optional-locks -c core.quotepath=false -c color.status=false status -unormal --branch --porcelain=2) - 17.889876ms
[DEBUG] 11:11:04.504 shell.go:HasFolder:435 → false
[TRACE] 11:11:04.504 shell.go:HasFolder(/Users/don/src/example.com/repo/.git/rebase-merge) - 16.468µs
[DEBUG] 11:11:04.504 shell.go:HasFolder:435 → false
[TRACE] 11:11:04.504 shell.go:HasFolder(/Users/don/src/example.com/repo/.git/rebase-apply) - 5.96µs
[DEBUG] 11:11:04.504 shell.go:HasFilesInDir:406 → false
[TRACE] 11:11:04.504 shell.go:HasFilesInDir(MERGE_MSG) - 13.719µs
[DEBUG] 11:11:04.504 shell.go:HasFilesInDir:406 → false
[TRACE] 11:11:04.504 shell.go:HasFilesInDir(CHERRY_PICK_HEAD) - 6.8µs
[DEBUG] 11:11:04.504 shell.go:HasFilesInDir:406 → false
[TRACE] 11:11:04.504 shell.go:HasFilesInDir(REVERT_HEAD) - 8.588µs
[DEBUG] 11:11:04.504 shell.go:HasFilesInDir:406 → false
[TRACE] 11:11:04.504 shell.go:HasFilesInDir(sequencer/todo) - 10.458µs
[TRACE] 11:11:04.504 shell.go:TemplateCache() - 307ns
[TRACE] 11:11:04.505 shell.go:TemplateCache() - 283ns
[ERROR] 11:11:04.505 text.go:Render:61 → template:  {{ url .Data.UpstreamIcon .Data.UpstreamURL}}{{ .Data.HEAD }}{{if .Data.BranchStatus }} {{ .Data.BranchStatus }}{{ end }}{{ if .Data.Working.Changed }}  {{ .Data.Working.String }}{{ end }}{{ if and (.Data.Working.Changed) (.Data.Staging.Changed) }} |{{ end }}{{ if .Data.Staging.Changed }}  {{ .Data.Staging.String }}{{ end }}{{ if gt .Data.StashCount 0 }}  {{ .Data.StashCount }}{{ end }} :1:4: executing " {{ url .Data.UpstreamIcon .Data.UpstreamURL}}{{ .Data.HEAD }}{{if .Data.BranchStatus }} {{ .Data.BranchStatus }}{{ end }}{{ if .Data.Working.Changed }} \uf044 {{ .Data.Working.String }}{{ end }}{{ if and (.Data.Working.Changed) (.Data.Staging.Changed) }} |{{ end }}{{ if .Data.Staging.Changed }} \uf046 {{ .Data.Staging.String }}{{ end }}{{ if gt .Data.StashCount 0 }} \uf692 {{ .Data.StashCount }}{{ end }} " at <url .Data.UpstreamIcon .Data.UpstreamURL>: error calling url: parse "https://_gitea@src.example.com:user/repo": invalid port ":user" after host
[TRACE] 11:11:04.505 shell.go:Shell() - 369ns
[TRACE] 11:11:04.506 shell.go:TemplateCache() - 269ns
[TRACE] 11:11:04.507 shell.go:TemplateCache() - 218ns
[TRACE] 11:11:04.507 shell.go:TemplateCache() - 217ns
[TRACE] 11:11:04.507 shell.go:TemplateCache() - 202ns
[DEBUG] 11:11:04.507 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.507 shell.go:Pwd() - 11.448µs
[TRACE] 11:11:04.507 shell.go:GOOS() - 111ns
[TRACE] 11:11:04.507 shell.go:ExecutionTime() - 139ns
[TRACE] 11:11:04.507 shell.go:TemplateCache() - 119ns
[TRACE] 11:11:04.508 shell.go:TemplateCache() - 202ns
[TRACE] 11:11:04.508 shell.go:Shell() - 144ns
[TRACE] 11:11:04.508 shell.go:Flags() - 94ns
[DEBUG] 11:11:04.508 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.508 shell.go:Pwd() - 4.402µs
[TRACE] 11:11:04.508 shell.go:GOOS() - 78ns
[DEBUG] 11:11:04.508 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.508 shell.go:Getenv(WSL_DISTRO_NAME) - 3.102µs
[TRACE] 11:11:04.508 shell_unix.go:IsWsl() - 6.174µs
[DEBUG] 11:11:04.525 shell.go:RunCommand:526 ↓
    Now drawing from 'AC Power'
     -InternalBattery-0 (id=7012451)	80%; AC attached; not charging present: true
[TRACE] 11:11:04.525 shell.go:RunCommand(pmset -g batt) - 17.451987ms
[TRACE] 11:11:04.525 shell_darwin.go:BatteryState() - 17.532847ms
[TRACE] 11:11:04.525 shell.go:TemplateCache() - 176ns
[TRACE] 11:11:04.526 shell.go:TemplateCache() - 262ns
[TRACE] 11:11:04.526 shell.go:Shell() - 410ns
[TRACE] 11:11:04.526 shell.go:TemplateCache() - 165ns
[TRACE] 11:11:04.526 shell.go:TemplateCache() - 215ns
[TRACE] 11:11:04.526 shell.go:TemplateCache() - 206ns
[DEBUG] 11:11:04.526 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.527 shell.go:Pwd() - 11.062µs
[TRACE] 11:11:04.527 shell.go:GOOS() - 92ns
[DEBUG] 11:11:04.527 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.527 shell.go:Getenv(AWS_VAULT) - 3.529µs
[DEBUG] 11:11:04.527 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.527 shell.go:Getenv(AWS_PROFILE) - 2.67µs
[DEBUG] 11:11:04.527 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.527 shell.go:Getenv(AWS_REGION) - 2.56µs
[DEBUG] 11:11:04.527 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.527 shell.go:Getenv(AWS_DEFAULT_REGION) - 16.802µs
[DEBUG] 11:11:04.527 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.527 shell.go:Getenv(AWS_CONFIG_FILE) - 2.386µs
[ERROR] 11:11:04.527 shell.go:FileContent:460 → open /Users/don/.aws/config: no such file or directory
[TRACE] 11:11:04.527 shell.go:FileContent(/Users/don/.aws/config) - 27.298µs
[DEBUG] 11:11:04.527 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.527 shell.go:Pwd() - 3.527µs
[TRACE] 11:11:04.527 shell.go:GOOS() - 60ns
[DEBUG] 11:11:04.527 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.527 shell.go:Getenv(AZURE_CONFIG_DIR) - 2.611µs
[DEBUG] 11:11:04.527 shell.go:HasFilesInDir:406 → true
[TRACE] 11:11:04.527 shell.go:HasFilesInDir(azureProfile.json) - 23.426µs
[DEBUG] 11:11:04.527 shell.go:FileContent:464 → {"installationId": "3661dd78-37e2-11ec-98cd-acde48001122", "subscriptions": []}
[TRACE] 11:11:04.527 shell.go:FileContent(/Users/don/.azure/azureProfile.json) - 58.469µs
[DEBUG] 11:11:04.527 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.527 shell.go:Getenv(POSH_AZURE_SUBSCRIPTION) - 4.198µs
[DEBUG] 11:11:04.527 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.527 shell.go:Pwd() - 5.828µs
[TRACE] 11:11:04.527 shell.go:GOOS() - 78ns
[DEBUG] 11:11:04.527 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.527 shell.go:Pwd() - 3.17µs
[DEBUG] 11:11:04.527 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.527 shell.go:Pwd() - 2.799µs
[TRACE] 11:11:04.527 shell.go:HasFiles(host.json) - 14.671µs
[DEBUG] 11:11:04.527 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.527 shell.go:Pwd() - 2.928µs
[TRACE] 11:11:04.527 shell.go:HasFiles(local.settings.json) - 9.565µs
[DEBUG] 11:11:04.527 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.527 shell.go:Pwd() - 2.919µs
[TRACE] 11:11:04.527 shell.go:HasFiles(function.json) - 9.62µs
[DEBUG] 11:11:04.527 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.527 shell.go:Pwd() - 3.067µs
[TRACE] 11:11:04.527 shell.go:GOOS() - 54ns
[DEBUG] 11:11:04.527 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.527 shell.go:Pwd() - 4.278µs
[DEBUG] 11:11:04.527 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.527 shell.go:Pwd() - 2.795µs
[TRACE] 11:11:04.527 shell.go:HasFiles(*.go) - 137.976µs
[DEBUG] 11:11:04.527 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.527 shell.go:Pwd() - 3.452µs
[TRACE] 11:11:04.527 shell.go:HasFiles(go.mod) - 12.592µs
[DEBUG] 11:11:04.527 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.527 shell.go:Pwd() - 3.028µs
[TRACE] 11:11:04.527 shell.go:GOOS() - 54ns
[DEBUG] 11:11:04.527 shell.go:CommandPath:547 → /usr/local/bin/kubectl
[TRACE] 11:11:04.527 shell.go:CommandPath(kubectl) - 71.502µs
[TRACE] 11:11:04.527 shell.go:HasCommand(kubectl) - 75.241µs
[ERROR] 11:11:04.576 shell.go:RunCommand:524 → exit status 1
[DEBUG] 11:11:04.576 shell.go:RunCommand:526 ↓
    error: current-context must exist in order to minify

[TRACE] 11:11:04.576 shell.go:RunCommand(kubectl config view --output yaml --minify) - 48.676804ms
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 12.03µs
[TRACE] 11:11:04.576 shell.go:GOOS() - 78ns
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 3.146µs
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 2.413µs
[TRACE] 11:11:04.576 shell.go:HasFiles(.perl-version) - 17.271µs
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 2.494µs
[TRACE] 11:11:04.576 shell.go:HasFiles(*.pl) - 114.925µs
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 3.306µs
[TRACE] 11:11:04.576 shell.go:HasFiles(*.pm) - 53.941µs
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 5.544µs
[TRACE] 11:11:04.576 shell.go:HasFiles(*.t) - 54.669µs
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 3.034µs
[TRACE] 11:11:04.576 shell.go:GOOS() - 71ns
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 2.48µs
[DEBUG] 11:11:04.576 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.576 shell.go:Getenv(VIRTUAL_ENV) - 5.014µs
[DEBUG] 11:11:04.576 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.576 shell.go:Getenv(CONDA_ENV_PATH) - 2.088µs
[DEBUG] 11:11:04.576 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.576 shell.go:Getenv(CONDA_DEFAULT_ENV) - 2.172µs
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 9.779µs
[TRACE] 11:11:04.576 shell.go:GOOS() - 41ns
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 2.401µs
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 2.39µs
[TRACE] 11:11:04.576 shell.go:HasFiles(*.rb) - 64.984µs
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 2.762µs
[TRACE] 11:11:04.576 shell.go:HasFiles(Rakefile) - 10.087µs
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 2.434µs
[TRACE] 11:11:04.576 shell.go:HasFiles(Gemfile) - 8.135µs
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 2.502µs
[TRACE] 11:11:04.576 shell.go:GOOS() - 54ns
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 2.617µs
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 2.38µs
[TRACE] 11:11:04.576 shell.go:HasFiles(*.rs) - 54.588µs
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 2.689µs
[TRACE] 11:11:04.576 shell.go:HasFiles(Cargo.toml) - 12.34µs
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 4.178µs
[TRACE] 11:11:04.576 shell.go:HasFiles(Cargo.lock) - 9.383µs
[DEBUG] 11:11:04.576 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.576 shell.go:Pwd() - 2.643µs
[TRACE] 11:11:04.576 shell.go:GOOS() - 77ns
[TRACE] 11:11:04.576 shell.go:TemplateCache() - 183ns
[TRACE] 11:11:04.577 shell.go:TemplateCache() - 172ns
[TRACE] 11:11:04.577 shell.go:Shell() - 294ns
[TRACE] 11:11:04.577 shell.go:Flags() - 12.327µs
[DEBUG] 11:11:04.577 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.578 shell.go:Pwd() - 22.86µs
[TRACE] 11:11:04.578 shell.go:GOOS() - 83ns
[TRACE] 11:11:04.578 shell.go:TemplateCache() - 146ns
[TRACE] 11:11:04.578 shell.go:Shell() - 139ns
[DEBUG] 11:11:04.578 shell.go:Pwd:355 → /Users/don/src/example.com/repo
[TRACE] 11:11:04.578 shell.go:Pwd() - 3.76µs
[TRACE] 11:11:04.578 shell.go:GOOS() - 42ns
[TRACE] 11:11:04.578 shell.go:ErrorCode() - 96ns
[TRACE] 11:11:04.578 shell.go:TemplateCache() - 95ns
[TRACE] 11:11:04.578 shell.go:Shell() - 70ns
[TRACE] 11:11:04.578 shell.go:TemplateCache() - 188ns
[DEBUG] 11:11:04.578 shell.go:Getenv:347 → NO DATA
[TRACE] 11:11:04.578 shell.go:Getenv(XDG_CACHE_HOME) - 3.985µs
[TRACE] 11:11:04.578 shell_unix.go:CachePath() - 29.658µs
[TRACE] 11:11:04.578 shell.go:Flags() - 77ns
```

--- a/src/segments/git.go
+++ b/src/segments/git.go
@@ -396,8 +396,8 @@ func (g *Git) getUpstreamIcon() string {
 		}
 		url = strings.TrimPrefix(url, "ssh://")
 		url = strings.TrimPrefix(url, "git://")
-		if strings.HasPrefix(url, "git@") {
-			url = strings.TrimPrefix(url, "git@")
+		if i := strings.Index(url, "@"); i >= 0 {
+			url = url[i+1:]
 			url = strings.Replace(url, ":", "/", 1)
 		}
 		url = strings.TrimSuffix(url, ".git")

--- a/src/segments/git_test.go
+++ b/src/segments/git_test.go
@@ -597,6 +597,7 @@ func TestGitUpstream(t *testing.T) {
 	}{
 		{Case: "No upstream", Expected: "", Upstream: ""},
 		{Case: "SSH url", Expected: "G", Upstream: "ssh://git@git.my.domain:3001/ADIX7/dotconfig.git"},
+		{Case: "Gitea", Expected: "EX", Upstream: "_gitea@src.example.com:user/repo.git"},
 		{Case: "GitHub", Expected: "GH", Upstream: "github.com/test"},
 		{Case: "Gitlab", Expected: "GL", Upstream: "gitlab.com/test"},
 		{Case: "Bitbucket", Expected: "BB", Upstream: "bitbucket.org/test"},
@@ -619,6 +620,7 @@ func TestGitUpstream(t *testing.T) {
 			GitIcon:         "G",
 			UpstreamIcons: map[string]string{
 				"mycustom.server": "CU",
+				"src.example.com": "EX",
 			},
 		}
 		g := &Git{

